app/models/rbac.rb

```diff
@@ -275,16 +271,23 @@ def self.accessible_tenant_ids_strategy(klass)
     TENANT_ACCESS_STRATEGY[klass.base_model.to_s]
   end
 
-  def self.find_targets_with_rbac(klass, scope, rbac_filters, find_options = {}, user_or_group = nil)
-    find_options = find_options_for_tenant(klass, user_or_group, find_options) if klass.respond_to?(:scope_by_tenant?) && klass.scope_by_tenant?
+  def self.find_targets_with_rbac(klass, scope, rbac_filters, find_options, user, miq_group)
+    if klass.respond_to?(:scope_by_tenant?) && klass.scope_by_tenant?
+      find_options = find_options_for_tenant(klass, user, miq_group, find_options)
+    end
 
-    return find_targets_with_direct_rbac(klass, scope, rbac_filters, find_options, user_or_group)     if apply_rbac_to_class?(klass)
-    return find_targets_with_indirect_rbac(klass, scope, rbac_filters, find_options, user_or_group)   if apply_rbac_to_associated_class?(klass)
-    return find_targets_with_user_group_rbac(klass, scope, rbac_filters, find_options, user_or_group) if apply_user_group_rbac_to_class?(klass)
-    find_targets_without_rbac(klass, scope, find_options)
+    if apply_rbac_to_class?(klass)
+      find_targets_with_direct_rbac(klass, scope, rbac_filters, find_options, user, miq_group)
+    elsif apply_rbac_to_associated_class?(klass)
+      find_targets_with_indirect_rbac(klass, scope, rbac_filters, find_options, user, miq_group)
+    elsif apply_user_group_rbac_to_class?(klass) && (user.try!(:self_service?) || miq_group.try!(:self_service?))
```

Comment:
thanks for clarification.
I misread - the statement was to move into `apply_user_grou_rbac_to_class?` NOT to move it back into `find_targets`